### DISCUSSION TITLE
Does AMP actually work with NextJS?

### DISCUSSION THREAD

**Discussion by willvlad** [vercel/next.js#23124] [Help]
Adding to index.js
`export const config = { amp: true };`

leads to

```
TypeError: Cannot read property '__html' of undefined
    at eval (webpack-internal:///./node_modules/next/dist/pages/_document.js:494:76)
```

Any idea why this is happening and what to do about it?

Our application is reasonably complex, using JS, CSS, not just an html blog. Are we not able to use AMP?

Using  "next": "^10.0.5",

**Comment by leerob** [comment]
Yes, it does! You can view an example here https://github.com/vercel/next.js/tree/canary/examples/amp.

**Comment by willvlad** [comment]
I saw that, but this bare-bones example isn't clear on several counts: 
1) which version of nextjs it works on - just canary or stable
2) does it work with anything of any complexity like menus, forms, etc or just vanilla html
3) most importantly why is it not working on our web app and has anyone managed to AMPify a proper NextJS site with menus and forms?

**Reply by leerob** [reply]
1. All versions
2. Yes, it works with all AMP features. We use AMP on vercel.com
3. I'm not sure about your app unless you provide a reproducible example.
